mri/operators/fourier/utils.py

```diff
@@ -150,7 +150,8 @@ def get_stacks_fourier(kspace_loc, volume_shape):
 
     try:
         idx_mask_z = np.asarray([
-            np.where(x == full_stack_z_loc)[0][0] for x in sampled_stack_z_loc
+            np.where(np.isclose(x, full_stack_z_loc))[0][0]
```

Comment:
why was this changed ?

Reply:
x == full_stack_z_loc does not work always, as seen by @mathrip .
Sometimes, we have issues just because they are approximately equal and not exactly equal.

Reply:
Why is not always equal ?
I mean how are they generated ?
I have a presupposition that is visibly wrong about how these stack locations are generated.

Reply:
So the way we generate is : 
**Expected:**  By linspace, we know we have Nz kspace z_lines, we expect them to be on a grid.
**Actual:** This is obtained from a function coming in from SPARKLING code that actually estimates where the kspace z_line is at based on what was run on the scanner, this is actually more accurate. (Note that we limit the discrete steps while running on scanner an there also could be some minor shifts). 

All we do here is to actually find the undersampled kspace locations based on what is sampled vs what should have been sampled.


So in reality, if we know that it is limited to a grid, with minor change in locations, we can still use this way of gridding rather than 3D

Reply:
ok makes sense